ui/src/dashboards/components/GraphOptionsCustomizableField.tsx

```diff
@@ -1,70 +1,186 @@
+
+    // Dragging downwards
+    if (dragIndex < hoverIndex && hoverClientY < hoverMiddleY) {
+      return
+    }
+
+    // Dragging upwards
+    if (dragIndex > hoverIndex && hoverClientY > hoverMiddleY) {
+      return
+    }
+    // Time to actually perform the action
+    props.moveField(dragIndex, hoverIndex)
+
+    // Note: we're mutating the monitor item here!
+    // Generally it's better to avoid mutations,
+    // but it's good here for the sake of performance
+    // to avoid expensive index searches.
+    monitor.getItem().index = hoverIndex
+  },
+}
+
+function MyDropTarget(dropv1, dropv2, dropfunc1) {
+  return target => DropTarget(dropv1, dropv2, dropfunc1)(target) as any
+}
+
+function MyDragSource(dragv1, dragv2, dragfunc1) {
+  return target => DragSource(dragv1, dragv2, dragfunc1)(target) as any
+}
+
+@MyDropTarget(FieldType, fieldTarget, (connect: DropTargetConnector) => ({
+  connectDropTarget: connect.dropTarget(),
+}))
+@MyDragSource(
+  FieldType,
+  fieldSource,
+  (connect: DragSourceConnector, monitor: DragSourceMonitor) => ({
+    connectDragSource: connect.dragSource(),
+    connectDragPreview: connect.dragPreview(),
+    isDragging: monitor.isDragging(),
+  })
+)
+export default class GraphOptionsCustomizableField extends PureComponent<
+  FieldSourceProps
+> {
   constructor(props) {
     super(props)
 
     this.handleFieldRename = this.handleFieldRename.bind(this)
     this.handleToggleVisible = this.handleToggleVisible.bind(this)
   }
+  public render(): JSX.Element | null {
+    const {
+      internalName,
+      displayName,
+      isDragging,
+      connectDragSource,
+      connectDragPreview,
+      connectDropTarget,
+      visible,
+    } = this.props
+
+    const fieldClass = `customizable-field${isDragging ? ' dragging' : ''}`
+    const labelClass = `${
+      visible
+        ? 'customizable-field--label'
+        : 'customizable-field--label__hidden'
+    }`
 
-  public render() {
-    const {internalName, displayName, visible} = this.props
-
-    return (
-      <div className="customizable-field">
-        <div
-          className={
-            visible
-              ? 'customizable-field--label'
-              : 'customizable-field--label__hidden'
-          }
-          onClick={this.handleToggleVisible}
-          title={
-            visible
-              ? `Click to HIDE ${internalName}`
-              : `Click to SHOW ${internalName}`
-          }
-        >
-          <span className={visible ? 'icon eye-open' : 'icon eye-closed'} />
-          {internalName}
+    const visibilityTitle = visible
+      ? `Click to HIDE ${internalName}`
+      : `Click to SHOW ${internalName}`
+
+    return connectDragPreview(
+      connectDropTarget(
+        <div className={fieldClass}>
+          <div className={labelClass}>
+            {connectDragSource(
+              <div className="customizable-field--drag">
+                <span className="hamburger" />
```

Comment:
this class name doesnt seem inline with other classes used throughout the application

Reply:
but it's a hamburger icon!! :))